WcaOnRails/app/models/user.rb

```diff
@@ -327,6 +327,34 @@ def avatar_requires_wca_id
     end
   end
 
+  after_save :changed_delegate_status
+  private def changed_delegate_status
+    user = User.find_by_wca_id(wca_id)
+    if saved_change_to_delegate_status? && !delegate_status
+      senior_delegate = User.find_by_id(senior_delegate_id_before_last_save)
+      DelegateStatusChangeMailer.notify_board_and_wqac_of_delegate_status_change(user, senior_delegate).deliver_later
+    elsif saved_change_to_delegate_status? && delegate_status
+      senior_delegate = user.senior_delegate
+      DelegateStatusChangeMailer.notify_board_and_wqac_of_delegate_status_change(user, senior_delegate).deliver_later
+    end
+  end
+
+  def delegate_status_name
+    return "Candidate Delegate" if delegate_status == "candidate_delegate"
+
+    return "Delegate" if delegate_status == "delegate"
+
+    return "Senior Delegate" if delegate_status == "senior_delegate"
+  end
+
+  def delegate_status_was_name
```

Comment:
As I commented above in `delegate_status_name`, you should not need this here.

Reply:
Indeed